src/dist/dist.rs

```diff
@@ -717,11 +727,48 @@ fn try_update_from_dist_<'a>(
                 None => Vec::new(),
             };
 
-            let changes = Changes {
+            let mut changes = Changes {
                 explicit_add_components: profile_components,
                 remove_components: Vec::new(),
             };
 
+            let mut not_found_components = Vec::new();
+
+            for component in components {
+                if let Some(component) =
+                    m.find_targeted_component(component, &toolchain.target, &toolchain.target)?
+                {
+                    changes.explicit_add_components.push(component);
+                } else {
+                    use crate::dist::manifest::Component;
+                    not_found_components.push(Component::new(component.to_string(), None, false));
+                }
+            }
+
+            for target in targets {
+                let triple = TargetTriple::new(target);
+                if let Some(component) =
+                    m.find_targeted_component("rust-std", &toolchain.target, &triple)?
```

Comment:
Is `rustc` sufficient, or do we really want `rust-std`?

Reply:
A target *is* `rust-std` for the given target, installed in the `host` toolchain.